nbhttp/websocket/upgrader.go

```diff
@@ -518,49 +519,49 @@ func (u *Upgrader) handleWsMessage(c *Conn, opcode MessageType, data []byte) {
 	}
 }
 
-func (u *connState) nextFrame() (opcode MessageType, body []byte, ok, fin, res1, res2, res3 bool) {
-	l := int64(len(u.buffer))
+func (wr *wsReader) nextFrame() (opcode MessageType, body []byte, ok, fin, res1, res2, res3 bool) {
+	l := int64(len(wr.buffer))
 	headLen := int64(2)
 	if l >= 2 {
-		opcode = MessageType(u.buffer[0] & 0xF)
-		res1 = int8(u.buffer[0]&0x40) != 0
-		res2 = int8(u.buffer[0]&0x20) != 0
-		res3 = int8(u.buffer[0]&0x10) != 0
-		fin = ((u.buffer[0] & 0x80) != 0)
-		payloadLen := u.buffer[1] & 0x7F
+		opcode = MessageType(wr.buffer[0] & 0xF)
+		res1 = int8(wr.buffer[0]&0x40) != 0
+		res2 = int8(wr.buffer[0]&0x20) != 0
+		res3 = int8(wr.buffer[0]&0x10) != 0
+		fin = ((wr.buffer[0] & 0x80) != 0)
+		payloadLen := wr.buffer[1] & 0x7F
 		bodyLen := int64(-1)
 
 		switch payloadLen {
 		case 126:
```

Comment:
good idea to define named constants for 126 and 127 with comment

Reply:
the RFC does not define names for these different numbers of size and it's really hard to give them names. gorilla/websocket use raw number either.